File: CSHTML5.Samples.Showcase/Other/MaterialDesign_Styles_Kit/DoubleToCornerRadiusConverter.cs
using System;
using Windows.UI.Xaml;
using Windows.UI.Xaml.Data;

namespace MaterialDesign_Styles_Kit
{
    public class DoubleToCornerRadiusConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            if (value is double)
            {
                return new CornerRadius((double)value);
            }
            return value;
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            throw new NotImplementedException();
        }
    }
}

File: src/Other/MaterialDesign_Styles_Kit/DoubleToCornerRadiusConverter.fs
namespace MaterialDesign_Styles_Kit

open System
#if SLMIGRATION
open System.Windows
open System.Windows.Controls
open System.Windows.Data
open System.Globalization
#else
open Windows.UI.Xaml
open Windows.UI.Xaml.Data
#endif

type DoubleToCornerRadiusConverter() =
    interface IValueConverter with
        member this.Convert(value, targetType, parameter, language) =
            match value with
            | :? double as d -> CornerRadius(d) :> obj
            | _ -> value

        member this.ConvertBack(value, targetType, parameter, language) =
            raise (NotImplementedException() :> exn)
